### PR TITLE
Feature/hmac

### PR DESCRIPTION
**Pull request by ben-graffle**
add hmac auth header to webhook requests

### CODE REVIEW

File: Graffle.FlowEventProcessor/Program.cs [hunk 1]
```diff
@@ -37,36 +37,51 @@ static async Task Main(string[] args)
...
             }
-            
-            if(!maximumBlockScanRangeEnvVariable.HasValue || maximumBlockScanRangeEnvVariable <= 0 || maximumBlockScanRangeEnvVariable > 250){
+
+            if (!maximumBlockScanRangeEnvVariable.HasValue || maximumBlockScanRangeEnvVariable <= 0 || maximumBlockScanRangeEnvVariable > 250)
+            {
                 throw new Exception("Specify MaximumBlockScanRange environment variable between 1 and 250.");
             }
 
+            if (string.IsNullOrEmpty(hmacToken) || !TryParseBase64(hmacToken, out byte[] hmacBytes))
```
Comment (dakota-graffle) [inline Graffle.FlowEventProcessor/Program.cs]: This makes HMAC required correct? I would say check if the value exists and its Base64, but they dont have to have a Token

File: Graffle.FlowEventProcessor/Program.cs [hunk 2]
```diff
@@ -37,36 +37,51 @@ static async Task Main(string[] args)
...
             Console.WriteLine($"Setup Flow Client");
             var flowClientFactory = new FlowClientFactory(nodeName);
             var flowClient = flowClientFactory.CreateFlowClient();
             Console.WriteLine($"Setup Flow Client Complete");
-            var httpClient = new HttpClient();
+
+            var hmacHandler = new HMACDelegatingHandler(hmacBytes);
+            var httpClient = new HttpClient(hmacHandler);
+            httpClient.DefaultRequestHeaders.Add("x-graffle-company-id", Guid.Empty.ToString());
+            httpClient.DefaultRequestHeaders.Add("x-graffle-project-id", Guid.Empty.ToString());
```
Comment (dakota-graffle) [inline Graffle.FlowEventProcessor/Program.cs:85]: Addd to the ReadMe that we provided empty tokens so they know how to decode

**Review by dakota-graffle — CHANGES_REQUESTED** [review]

File: Graffle.FlowEventProcessor/Program.cs (hunk 1 above)
Comment (ben-graffle) [reply Graffle.FlowEventProcessor/Program.cs]: Made optional in commit [feb81ed](https://github.com/Graffle/flow-event-processor/pull/2/commits/feb81ed43fe98ca7867dd54d4322ddee9ff0140e)

File: Graffle.FlowEventProcessor/Program.cs (hunk 2 above)
Comment (ben-graffle) [reply Graffle.FlowEventProcessor/Program.cs:85]: Readme updated [ca6f561](https://github.com/Graffle/flow-event-processor/pull/2/commits/ca6f561980f21098abbadf4570ef9a1cce2875ad)

**Review by dakota-graffle — APPROVED** [review]

**Pull request merged**